internal/sighandler/awaiting.go

```diff
@@ -43,6 +46,7 @@ func (as *Awaiting) WaitForFunc(f func() error) error {
 
 	go func() {
 		defer close(errCh)
+		defer func() { panics.HandlePanics(recover(), debug.Stack()) }()
```

Comment:
It might be better to place the panic handling handling first. I'm not certain, but I think it should handle a panic if errCh is already closed somehow.